lib/sprinkler/devices/watering_policy.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Sprinkler.Devices.WateringPolicy do
+  # TODO: check this values
+  @soil_moisture_wet 70
+  @soil_moisture_medium_wet 50
+  @ambience_temperature_high 25
+  @humidity_high 50
+
+  def should_irrigate(moisture_input, temperature_input, humidity_input) do
```

Comment:
Since we are not returning a truthy value a predicate seems like a bad choice for this method's name

```suggestion
  def irrigation_amount(moisture_input, temperature_input, humidity_input) do
```

Reply:
You mention below that remembering the order of the arguments it's difficult and that the method needs refactoring, I agree completely. Sanding naked values to a function is not FP. How about encapsulating all this data?

```elixir
# lib/sprinkler/devices/gardening_reading.ex
defmodule Devices.GardenReading do
  defstruct [:moisture, :temperature, :humidity]
end
```

And then
```suggestion
  alias Devices.GardenReading

  def should_irrigate(%GardenReading{moisture: moisture_input, temperature: temperature_input, humidity: humidity_input}) do
```

Reply:
Looks good. Im having issues with test cases using this, how can I pass it to the test cases: `%GardenReading{
        moisture: moisture_input,
        temperature: temperature_input,
        humidity: humidity_input
      }`??